basics/uattributes.adoc

```diff
@@ -59,29 +59,57 @@ The following table defines attributes that are used for all message types:
 |`id`
 |link:uuid.adoc[UUID]
 |yes
-|A unique message identifier. The timestamp contained in the UUID is used as the message's creation time.
+a|
+[.specitem,oft-sid="req~up-attributes-id~1",oft-needs="impl,utest"]
+--
+Every message *MUST* have a unique message identifier. The timestamp contained in the UUID is used as the message's creation time.
+--
 
 |`priority`
 |link:qos.adoc[UPriority]
 |no
-|The QoS level that this message should be processed/delivered with. If not specified explicitly, the link:qos.adoc#default-priority[default priority level] is assumed.
-
+a|
+The QoS level that this message should be processed/delivered with. If not specified explicitly, the link:qos.adoc#default-priority[default priority level] is assumed.
+[.specitem,oft-sid="req~up-attributes-priority~1",oft-needs="impl,utest"]
+--
+* *MUST* have a priority attribute.
+--
 |`ttl`
 |`unsigned integer`
 |no
-a|The amount of time (in milliseconds) after which this message *MUST* no longer be delivered/processed. The message is considered expired if this attribute is set to a positive value and
+a|
+The amount of time (in milliseconds) after which this message is no longer delivered/processed.
+[.specitem,oft-sid="req~up-attributes-ttl~1",oft-needs="impl,utest"]
+--
+* *MUST* have a TTL attribute.
+--
+[.specitem,oft-sid="req~up-attributes-ttl-timeout~1",oft-needs="impl,utest"]
+--
+* The message *MUST* expire if the current time is greater than the sum of the message's creation time and the TTL value.
+--
 
 stem:[t_{current} > t_{id} + ttl]
 
 |`traceparent`
 |https://w3c.github.io/trace-context/#traceparent-header[W3C Trace Context Level 3, traceparent]
 |no
-|A tracing identifier to use for correlating messages across the system. Intended to be compatible with https://github.com/cloudevents/spec/blob/main/cloudevents/extensions/distributed-tracing.md[CloudEvents distributed tracing extension].
+a|
+A tracing identifier to use for correlating messages across the system. Intended to be compatible with https://github.com/cloudevents/spec/blob/main/cloudevents/extensions/distributed-tracing.md[CloudEvents distributed tracing extension].
+
+[.specitem,oft-sid="req~up-attributes-traceparent~1",oft-needs="impl,utest"]
+--
+* *MAY* contain traceparent attribute.
+--
+
 
 |`payload format`
 |link:upayloadformat.adoc[UPayloadFormat]
 |no
-|The format of the payload (data).
+a|The format of the payload (data).
+[.specitem,oft-sid="req~up-attributes-payload-format~1",oft-needs="impl,utest"]
+--
+* *MAY* contain payload format attribute.
```

Comment:
I would have assumed this is necessary, even if just to write that it's unspecified.

Also according to the .proto it's [not optional](https://github.com/eclipse-uprotocol/up-spec/blob/8801cc68be1356df6e861e56aa824fbdad853d54/up-core-api/uprotocol/v1/uattributes.proto#L66).

Reply:
It is optional, all message/enum typed fields in proto3 are per se optional, i.e. they do not need to be marked with the `optional` modifier.
For an enum, if no value is set, the enum's default value is assumed.